### PR TITLE
PassWrapper: update for llvm/llvm-project@a3319371970b

### PR DESCRIPTION
**Pull request by durin42**
@rustbot label: +llvm-main

### CODE REVIEW

File: compiler/rustc_llvm/llvm-wrapper/PassWrapper.cpp [hunk 1]
```diff
@@ -451,14 +451,30 @@ extern "C" LLVMTargetMachineRef LLVMRustCreateTargetMachine(
...
+#else
     Options.CompressDebugSections = DebugCompressionType::None;
+#endif
   }
 
+#if LLVM_VERSION_GE(19, 0)
+  Options.MCOptions.X86RelaxRelocations = RelaxELFRelocations;
+#else
   Options.RelaxELFRelocations = RelaxELFRelocations;
+#endif
```
[the remark this answers is not in the document]
Comment (durin42) [reply compiler/rustc_llvm/llvm-wrapper/PassWrapper.cpp:477]: It....might be, but I'd have to think hard about it. It's readable to my brain to not have the indirection of the reference in the C++, so I think I'd rather leave it as-is.

[the remark this answers is not in the document]
Comment (workingjubilee) [reply compiler/rustc_llvm/llvm-wrapper/PassWrapper.cpp:477]: That's fine then. I find the current form a little challenging to read but I agree that it's closer to a wash.

[the remark this answers is not in the document]
Comment (cuviper) [reply compiler/rustc_llvm/llvm-wrapper/PassWrapper.cpp:477]: Another way to approach that is to assign the desired `DebugCompressionType` to a local and then have one conditional assignment block.

**Pull request merged**